jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractDeploymentService.java

```diff
@@ -176,19 +179,40 @@ public void commonDeploy(DeploymentUnit unit, DeployedUnitImpl deployedUnit, Run
     
     @Override
     public void undeploy(DeploymentUnit unit) {
-        List<Integer> states = new ArrayList<Integer>();
-        states.add(ProcessInstance.STATE_ACTIVE);
-        states.add(ProcessInstance.STATE_PENDING);
-        states.add(ProcessInstance.STATE_SUSPENDED);
-        Collection<ProcessInstanceDesc> activeProcesses = runtimeDataService.getProcessInstancesByDeploymentId(unit.getIdentifier(), states, new QueryContext());
-        if (!activeProcesses.isEmpty()) {
-            throw new IllegalStateException("Undeploy forbidden - there are active processes instances for deployment " 
-                                            + unit.getIdentifier());
-        }
+        undeploy(unit, false);
+    }
+
+    @Override
+    public void undeploy(DeploymentUnit unit, boolean abortInstances) {
+
         synchronized (this) {
             DeployedUnit deployed = deploymentsMap.remove(unit.getIdentifier());
+
             if (deployed != null) {
+
                 RuntimeManager manager = deployed.getRuntimeManager();
+
+                if(abortInstances) {
+                    List<Integer> states = new ArrayList<>();
```

Comment:
I think, this `states` list could be moved outside of this method to avoid repetitive initialization.